### PR TITLE
stdalgos (9/10): team-level API, and rst update for release 4.2

### PR DESCRIPTION
**Pull request by antoinemeyer5**
Add team-level API to and convert to rst the following algorithms:
```
Kokkos_IsSorted.hpp
Kokkos_MinElement.hpp
Kokkos_MaxElement.hpp
Kokkos_MinMaxElement.hpp
Kokkos_IsPartitioned.hpp
Kokkos_PartitionCopy.hpp
```

NOTE: this follows the new structure for documenting the API, but does not use sphinx because the lack of properly aligning things make things quite unreadable.

Open individual html with Firefox (must be firefox or it won't display correctly) 
[Archive.zip](https://github.com/kokkos/kokkos-core-wiki/files/13692197/Archive.zip)

### CODE REVIEW

File: docs/source/API/algorithms/std-algorithms/all/StdIsSorted.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Description
+-----------
+
+- (1,2,3,4): checks if the elements in the range ``[first, last)`` (1,2)
+  or in ``view`` (3,4) are sorted in non-descending order using ``operator<`` to compare two elements.
+
+- (5,6,7,8): checks if the elements in the range ``[first, last)`` (5,6)
+  or in ``view`` (7,8) are sorted in non-descending order using the binary functor ``comp`` to compare two elements.
+
```
Comment (fnrizzi) [inline docs/source/API/algorithms/std-algorithms/all/StdIsSorted.rst:15]: description should be more generic. 
```
Checks if the elements in a range or in a ``View`` are sorted in non-descending order using either ``operator<`` to compare two elements or a user-provided comparison operator

File: docs/source/API/algorithms/std-algorithms/all/StdIsPartitioned.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+     bool operator()(const ValueType & operand) const {
+       constexpr auto zero = static_cast<ValueType>(0);
+       return (operand < zero);
+     }
+   };
+
+   using view_type = Kokkos::View<int*>;
+   view_type a("a", 15);
+   // fill a somehow
+
```
Comment (ajpowelsnl) [inline docs/source/API/algorithms/std-algorithms/all/StdIsPartitioned.rst:141]:
```suggestion
//TODO: write a tiny loop populating a 1D Kokkos View with a single integer value
```

File: docs/source/API/algorithms/std-algorithms/all/StdMaxElement.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- if ``view`` is empty, it returns ``Kokkos::Experimental::end(view)``.
+
+Example
+~~~~~~~
+
+.. code-block:: cpp
+
+   namespace KE = Kokkos::Experimental;
+   Kokkos::View<double*> a("a", 13);
+   // fill a somehow
```
Comment (ajpowelsnl) [inline docs/source/API/algorithms/std-algorithms/all/StdMaxElement.rst:131]:
```suggestion
   // TODO: write a tiny loop populating a 1D Kokkos View with a single integer value
```

File: docs/source/API/algorithms/std-algorithms/all/StdMinElement.rst [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+- if ``view`` is empty, it returns ``Kokkos::Experimental::end(view)``.
+
+Example
+~~~~~~~
+
+.. code-block:: cpp
+
+   namespace KE = Kokkos::Experimental;
+   Kokkos::View<double*> a("a", 13);
+   // fill a somehow
```
Comment (ajpowelsnl) [inline docs/source/API/algorithms/std-algorithms/all/StdMinElement.rst:160]:
```suggestion
   //TODO: write a tiny loop populating a 1D Kokkos View with a single integer value
```

**Review by ajpowelsnl — APPROVED** [review]
@antoinemeyer5 , @fnrizzi , @crtrott -- fine to merge the updates, but in the future, we should make sure that example code is paste-able and immediately run-able (assuming compilers in place), vs. pseudo code

**Review by crtrott — APPROVED** [review]

**Pull request merged**